ansible/roles/provision-vm/tasks/main.yml

```diff
@@ -0,0 +1,63 @@
+---
+
+#
+# Provisioning takes place once the VM is created. We need to handle a number
+# of platforms here, and the ordering of this logic is quite important
+# for the Container OS platforms like flatcar and fedora-coreos.
+#
+# For those platforms, Python does not exist (which is a requirement for 
+# ansible) so it is installed first, before the rest of the provisioning
+# continues.
+#
+
+# set facts for weird platforms that report different families
+- set_fact:
+    is_coreos: "{{ vm_config.find('coreos') != -1 }}"
+    is_garden: "{{ vm_config == 'garden-linux_garden-linux' }}"
+    is_flatcar: "{{ vm_config.find('flatcar') != -1 }}"
```

Comment:
I did try that but they were always strings unfortunately (I think due to the way yaml works rather than Ansible) so regardless, the `|bool` additions need to exist because otherwise it always evaluates to true since it's checking the 'truthy' value of a string (i.e. `"False" == True`)